test/Microsoft.AspNetCore.JsonPatch.Test/IntegrationTests/AnonymousObjectIntegrationTest.cs

```diff
@@ -0,0 +1,242 @@
+            };
+
+            // create patch
+            var patchDocument = new JsonPatchDocument();
+            patchDocument.Add("Nested/NewInt", 1);
+
+            var serialized = JsonConvert.SerializeObject(patchDocument);
+            var deserialized = JsonConvert.DeserializeObject<JsonPatchDocument>(serialized);
+
+            var exception = Assert.Throws<JsonPatchException>(() =>
+            {
+                deserialized.ApplyTo(targetObject);
+            });
+            Assert.Equal(
+                string.Format("The target location specified by path segment '{0}' was not found.", "NewInt"),
+                exception.Message);
+        }
+
+        [Fact]
+        public void Add_DoesNotReplace()
+        {
+            var targetObject = new
+            {
+                StringProperty = "A"
+            };
+
+            // create patch
+            var patchDocument = new JsonPatchDocument();
+            patchDocument.Add("StringProperty", "B");
+
+            var serialized = JsonConvert.SerializeObject(patchDocument);
+            var deserialized = JsonConvert.DeserializeObject<JsonPatchDocument>(serialized);
+
+            var exception = Assert.Throws<JsonPatchException>(() =>
+            {
+                deserialized.ApplyTo(targetObject);
+            });
+            Assert.Equal(
+                string.Format("The property at path '{0}' could not be updated.", "StringProperty"),
+                exception.Message);
+        }
+
+        [Fact]
+        public void RemoveProperty_ShouldFail_IfRootIsAnonymous()
+        {
+            dynamic targetObject = new
+            {
+                Test = 1
+            };
+
+            // create patch
+            var patchDocument = new JsonPatchDocument();
+            patchDocument.Remove("Test");
+
+            var serialized = JsonConvert.SerializeObject(patchDocument);
+            var deserialized = JsonConvert.DeserializeObject<JsonPatchDocument>(serialized);
+
+            var exception = Assert.Throws<JsonPatchException>(() =>
+            {
+                deserialized.ApplyTo(targetObject);
+            });
+            Assert.Equal(
+                string.Format("The property at path '{0}' could not be updated.", "Test"),
+                exception.Message);
+        }
+
+        [Fact]
+        public void ReplaceGuid_ExpandoObject_InAnonymous()
+        {
+            dynamic nestedObject = new ExpandoObject();
+            nestedObject.GuidValue = Guid.NewGuid();
+
+            dynamic targetObject = new
+            {
+                NestedObject = nestedObject
+            };
+
+            var newGuid = Guid.NewGuid();
+            // create patch
+            var patchDocument = new JsonPatchDocument();
+            patchDocument.Replace("nestedobject/GuidValue", newGuid);
+
+            patchDocument.ApplyTo(targetObject);
+
+            Assert.Equal(newGuid, targetObject.NestedObject.GuidValue);
+        }
```

Comment:
I think that most of these tests don't add unique value if the **pivot** in this case is anonymously-typed objects. anon-t objects behave like POCO's with `get; private set;` properties. 

It seems like we need 4 tests here.
1. We can get a property's value
2. We cannot set a property's value
3. We cannot add a property
4. We cannot remove a property